src/orange/detail/OrientedBoundingZone.cc

```diff
@@ -0,0 +1,190 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/detail/OrientedBoundingZone.cc
+//---------------------------------------------------------------------------//
+#include "OrientedBoundingZone.hh"
+
+#include <numeric>
+
+#include "orange/transform/TransformVisitor.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct from inner/outer half-widths and a corresponding transform.
+ */
+OrientedBoundingZone::OrientedBoundingZone(Real3Id inner_hw_id,
+                                           Real3Id outer_hw_id,
+                                           TransformId transform_id,
+                                           Storage* storage)
+    : inner_hw_id_(inner_hw_id)
+    , outer_hw_id_(outer_hw_id)
+    , transform_id_(transform_id)
+    , storage_(storage)
+{
+    CELER_EXPECT(storage_);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Calculate the safety distance for a position inside the outer box.
+ *
+ * There are two cases:
+ *
+ * Case 1: the point is between the inner and outer boxes, resulting in a
+ * safety distance of zero.
+ *
+ * Case 2: the point is inside both the inner and outer boxes, in which case
+ * the safety distance is the minimum distance from the given point to any
+ * point on the outer box. This is calculated by finding in the minimum of the
+ * distances to each half width.
+ */
+real_type OrientedBoundingZone::safety_distance_inside(Real3 pos)
+{
+    CELER_EXPECT(this->is_inside_outer(pos));
+
+    auto trans_pos = this->quadrant_zero(this->translate(pos));
+    auto inner_hw = storage_->half_widths[inner_hw_id_];
+
+    if (!this->is_inside(trans_pos, inner_hw))
+    {
+        return 0.;
```

Comment:
Avoid double-precision literals
```suggestion
        return 0;
```